lib/modules/dosomething/dosomething_reportback/includes/ReportbackItemTransformer.php

```diff
@@ -128,4 +129,29 @@ private function removeUnauthorizedStatuses($statuses) {
     return $statuses;
   }
 
+  /**
+   * Remove Reportback Items with unauthorized statuses based on user permission.
+   *
+   * @param  array $data
+   * @return mixed
+   * @throws Exception
+   */
+  private function removeUnauthorizedResults($data) {
```

Comment:
Ideally yes, which is what is happening in the index collection call. But in a retrieve where you hit something like `/reportback-items/1234` you won't know ahead of time if that item is flagged or not since you're specifying a specific one.